src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -237,6 +228,25 @@ else if (shouldStartLine) {
     }
 
     /**
+     * Checks that a right curly should be alone on a line.
+     * @param details Details for validation
+     * @param bracePolicy option for placing the right curly brace
+     * @return true if a right curly should be alone on a line.
+     */
+    private static boolean shouldBeAloneOnLine(Details details, RightCurlyOption bracePolicy) {
+        final boolean alone = bracePolicy == RightCurlyOption.ALONE
+                && !isAloneOnLine(details)
+                && !isEmptyBody(details.lcurly);
+        final boolean aloneOrSingleline = alone
```

Comment:
This has introduced new violation:
>UnnecessaryLocalVariable
Reports unnecessary local variables, which add nothing to the comprehensibility of a method. Variables caught include local variables which are immediately returned, local variables that are immediately assigned to another variable and then not used, and local variables which always have the same value as another local variable or parameter.